### ISSUE TITLE
SwiftWasm 5.5 release

### ISSUE THREAD

**Issue by MaxDesiatov** [swiftwasm/swift#3604]
Here's a preliminary list of things that we'd probably want to see completed for the release:

- [x] Fix SwiftPM resources broken in 5.5 https://github.com/apple/swift-package-manager/pull/3819
- [x] Update `carton` for SwiftWasm 5.5 https://github.com/swiftwasm/carton/pull/263
- [x] Try to implement some `async`/`await` APIs from fresh SwiftUI in Tokamak with a JSKit version from the point above to give it more exposure and testing.
- [ ] Set up at least one build agent for Linux on ARM64 for toolchain/SDK builds. I guess Amazon Linux is the easiest one since AWS has ARM64 hosts. We could also set up Ubuntu 20.04 as well if we have capacity for that. Tracked separately as https://github.com/swiftwasm/swift/issues/1854
- [x] How do we support actors? Do `actor` modifiers just become no-op modifiers, since everything is running on the main thread anyway?
- [x] Make sure that `async`/`await` in JSKit is solid, see https://github.com/swiftwasm/JavaScriptKit/pull/137
- [x] https://github.com/swiftwasm/swift/issues/3646
- [x] https://github.com/swiftwasm/swift/issues/3645

@swiftwasm/toolchain-team WDYT? Anything else you think is required for a successful release? Do you think any of the points above are redundant and we should release ASAP as it is right now?

**Comment by kateinoigakukun** [comment]
IIRC, we need to cherry-pick my recent repairs of cooperative executor https://github.com/apple/swift/pull/39365 https://github.com/apple/swift/pull/39092
And also need to run `test/Concurrency/Runtime` test cases at least to ensure it works well.
If `actor` needs to be handled in wasm32 specially, some bugs will be found by the test cases.
(But as far as I understand, we don't need to handle them because only actor implementation (Actor.cpp) cares atomicity and job_enqueue procedure doesn't care it.)

**Comment by MaxDesiatov** [comment]
Thanks, I'm doing that in https://github.com/swiftwasm/swift/pull/3606

**Comment by MaxDesiatov** [comment]
Found another issue, does not look like a blocker to me as I can work around it in `carton` by passing flags there https://github.com/swiftwasm/swift/issues/3891

**Comment by MaxDesiatov** [comment]
`carton` is now ready for SwiftWasm 5.5, PR is waiting for approval here https://github.com/swiftwasm/carton/pull/263

**Comment by MaxDesiatov** [comment]
TWIMC, 5.5.0 distribution CI run started here https://github.com/swiftwasm/swift/actions/runs/1495522619

**Comment by MaxDesiatov** [comment]
Looks like there's a bug in distribution scripts, where intermittently the distribution name can have a certain timestamp, but the contents of it will have a timestamp for a previous day. I've triggered a new build in the 5.5 branch and I hope the newly produced distribution will be named correctly. Will tag 5.5.0 off that new distribution of that snapshot then.

Debugging the script and fixing it may take just as much time as simple re-triggering I've done. But if it's borked again tomorrow, guess I'll have to investigate this thoroughly and fix it.

**Comment by MaxDesiatov** [comment]
New release is tagged: https://github.com/swiftwasm/swift/releases/tag/swift-wasm-5.5.0-RELEASE, I'll keep it marked as pre-release until 27th of November, when everyone in the US is back from Thanksgiving holidays, which potentially will give us more coverage/shares when we announce it. Also gives us more time to test and tag new versions of Tokamak and carton.

I got it tagged off `wasm-5.5-SNAPSHOT-2021-11-16-a`. Later snapshots have their macOS installers (at least those for arm64) seemingly corrupted, but there's little to no difference between `wasm-5.5-SNAPSHOT-2021-11-16-a` and later snapshots. For distribution script to work without changes I had to delete later 5.5 tags though.
